hie-plugin-api/Haskell/Ide/Engine/Cradle.hs

```diff
@@ -378,6 +379,13 @@ cabalHelperCradle file = do
                                     }
                    }
     where
+      fixImportDirs :: FilePath -> String -> String
+      fixImportDirs base_dir arg =
+        if "-i" `isPrefixOf` arg
+          then let dir = drop 2 arg
+              in if isRelative dir then ("-i" ++ base_dir ++ "/" ++ dir)
```

Comment:
Not going to work on windows? You need to use `</>` from `filepath`.

Reply:
Then we probably ought to fix that in hie-bios as well ;D Since this is copy-pasted from there

Reply:
also, is missing an additional check: `not (null dir) && isRelative dir` since `isRelative "" == True`.